### PR TITLE
[ENHANCEMENT] #848 increase column width and page size

### LINKED ISSUE #848: Index Management UI: Name Width and Row Count
**Is your feature request related to a problem? Please describe.**

As an admin of our large OpenSearch cluster, I have been relying heavily on both the API and Dashboard UI as we migrate snapshotted data from older products. 

The Index Management UI has two unfortunate limitations which I'd like to suggest trivial improvements of:

1. Width of the `Index` column: Our migrated indices have names like `stable-testnet-v1.0-ilm-000065-migrated` or `stable-devnet-v1.0-ilm-000028-migrated-remapped`, which do not fit without truncation. A width of about 400px seems sufficient, though resizable columns would also be great.

image

2. There is no capability to show more than 50 rows. Given many years of "ISM" rollovers, we have many pages of indices. It would be excellent to provide a free form value, or perhaps set an allowed max value in "Advanced Settings". 

Note that OpenSearch 1.x a `size` query string parameter, which is easy to "hack". This no longer exists in OpenSearch 2.x.

image

**Describe the solution you'd like**

1. Allow resizing columns in Index Management data tables.
2. Or increased the fixed size of the "index" column to a more generous value.
3. Allow a larger maximum for number of rows display in Index Management.

**Describe alternatives you've considered**

Relying solely on the API, which is more time consuming, and may have extra security overhead.

**Additional context**

We're using the latest version that AWS Managed OpenSearch provides (2.7 currently), however I also checked https://playground.opensearch.org/, which has no change in this UI on OpenSearch 2.9.

### PR DESCRIPTION
**Pull request by kohinoor98**
### Description
1. Changed the Width of the Index column
2. Changed Page Size options for the Indices tab

### Issues Resolved
#848

### Check List
- [X] Commits are signed per the DCO using --signoff
- [X] Tested code for multiple index lengths

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/Indices/utils/constants.tsx [hunk 1]
```diff
@@ -45,7 +46,7 @@ const getColumns = (props: IColumnOptions): EuiTableFieldDataColumnType<ManagedC
       sortable: true,
       truncateText: false,
       textOnly: true,
-      width: "250px",
+      width: "320px",
```
Comment (bowenlan-amzn) [inline public/pages/Indices/utils/constants.tsx:49]: Would it be possible to let user drag the width of our table?
Currently although the text is capped, I remember user can hover on it and see the name which is fine.

Adding width here is not the best because people may have very long index name 😅

File: public/pages/Indices/containers/Indices/Indices.tsx [hunk 2]
```diff
@@ -211,25 +226,79 @@ export default class Indices extends Component<IndicesProps, IndicesState> {
...
 
-    const { history } = this.props;
+    const renderCellValue = ({ rowIndex, columnId }) => {
+      const rowData = this.state.indices[rowIndex];
+
+      console.log(rowData);
+
+      switch (columnId) {
+        case "index":
+          return <IndexDetail {...this.props} index={rowData.index} />;
```
Comment (bowenlan-amzn) [inline public/pages/Indices/containers/Indices/Indices.tsx:253]: Loading this IndexDetail component is the cause of endless call. @kohinoor98 can you reach out to @SuZhou-Joe the writter for some guidance here.

**Comment by kamingleung** [comment]
@kohinoor98 Would you mind posting screenshots or video capture on the pages that have been migrated to DataGrid component? Thanks.

File: public/pages/Indices/containers/Indices/Indices.tsx (hunk 2 above)
Comment (kohinoor98) [reply public/pages/Indices/containers/Indices/Indices.tsx:253]: Yup, will do. Thanks @bowenlan-amzn

File: public/pages/Indices/containers/Indices/Indices.tsx [hunk 3]
```diff
@@ -37,6 +40,7 @@ import { SECURITY_EXCEPTION_PREFIX } from "../../../../../server/utils/constants
 import IndicesActions from "../IndicesActions";
 import { destroyListener, EVENT_MAP, listenEvent } from "../../../../JobHandler";
 import "./index.scss";
+import IndexDetail from "../../../IndexDetail/containers/IndexDetail";
```
Comment (SuZhou-Joe) [inline public/pages/Indices/containers/Indices/Indices.tsx:43]:
```suggestion
import IndexDetail from "../../containers/IndexDetail";
```
You are using wrong component here.

Comment (SuZhou-Joe) [reply public/pages/Indices/containers/Indices/Indices.tsx:43]: The IndexDetail inside IndexDetail page is a super heavy component and should not be rendered inside a cell of data grid.

File: public/index.ts [hunk 4]
```diff
@@ -6,6 +6,7 @@
 import { PluginInitializerContext } from "opensearch-dashboards/public";
 import { IndexManagementPlugin } from "./plugin";
 import { Action, UIAction } from "../models/interfaces";
+import "@opensearch-project/oui/dist/oui_theme_light.css";
```
Comment (SuZhou-Joe) [inline public/index.ts:9]: The theme and its css should already be imported in OSD core, I do not think we should explicitly import theme.css in a plugin.

Comment (kohinoor-ravenri) [reply public/index.ts:9]: Yup @SuZhou-Joe. I got rid of `OUI`(faced issues with the library and documentation) entirely and switched to `EuiDataGrid`.
I am still working on the pagination part for the table. I will push the latest code with screenshots as I get some time to test for edge cases.

Thanks,
KC

Comment (kohinoor98) [reply public/index.ts:9]: Yup @SuZhou-Joe. I got rid of `OUI`(faced issues with the library and documentation) entirely and switched to `EuiDataGrid`.
I am still working on the pagination part for the table. I will push the latest code with screenshots as I get some time to test for edge cases.

Thanks,
KC
